Source/WebCore/page/Navigation.cpp

```diff
@@ -104,7 +104,7 @@ EventTargetInterface Navigation::eventTargetInterface() const
 Navigation::Result Navigation::navigate(const String& /* url */, NavigateOptions&&, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished)
 {
     // FIXME: keep track of promises to resolve later.
-    Ref entry = NavigationHistoryEntry::create(scriptExecutionContext(), { });
+    Ref entry = NavigationHistoryEntry::create(scriptExecutionContext(), { }, -1);
```

Comment:
-1 seems a bit magical. Would this be better as std::optional or some such?

Reply:
Good point, changed to use std::optional indeed.